src/test/fuzz/util.cpp

```diff
@@ -524,7 +525,28 @@ CNetAddr ConsumeNetAddr(FuzzedDataProvider& fuzzed_data_provider) noexcept
     } else if (network == Network::NET_INTERNAL) {
         net_addr.SetInternal(fuzzed_data_provider.ConsumeBytesAsString(32));
     } else if (network == Network::NET_ONION) {
-        net_addr.SetSpecial(fuzzed_data_provider.ConsumeBytesAsString(32));
+        auto pub_key{fuzzed_data_provider.ConsumeBytes<uint8_t>(torv3::ADDR_SIZE)};
+        if (pub_key.size() > 0 && pub_key[0] & 1) {
+            pub_key.resize(32);
+            // Create valid onion address to prevent the address checksum from
+            // being a fuzz blocker.
+            uint8_t check_sum[torv3::CHECKSUM_LEN];
+            torv3::Checksum({pub_key.data(), pub_key.size()}, check_sum);
+
+            std::vector<uint8_t> onion_addr;
+            onion_addr.insert(onion_addr.end(), pub_key.begin(), pub_key.end());
+            onion_addr.push_back(check_sum[0]);
+            onion_addr.push_back(check_sum[1]);
+            onion_addr.push_back(torv3::VERSION[0]);
+            assert(onion_addr.size() == torv3::TOTAL_LEN);
+
+            auto encoded{EncodeBase32({onion_addr.data(), onion_addr.size()})};
+            encoded += ".onion";
+            net_addr.SetSpecial(encoded);
```

Comment:
To ensure we actually did what we intended:
```suggestion
            const bool ok = net_addr.SetSpecial(encoded);
            assert(ok);
```